app/views/layouts/_adv_search_body.html.haml

```diff
@@ -79,7 +79,7 @@
             %label.control-label.col-md-5
               = _("Global search:")
             .col-md-6
-              - checked = @edit && @edit[:expression] && @edit[:expression][:selected] && @edit[:expression][:selected][:typ] == "global"
+              - checked = @edit && @edit[:expression] && @edit[:expression][:selected] && @edit[:search_type] == "global"
```

Comment:
.. OK, so.. apparently `default` is magic and can't be created..

So, I guess that `adv_search_toggle` code should save `"global"` when `"default"` (so that editing a `default` filter creates a new `global` one), and then the code in the haml can skip the `default` check :).